br/pkg/lightning/restore/restore.go

```diff
@@ -431,14 +437,25 @@ func (rc *Controller) Close() {
 }
 
 func (rc *Controller) Run(ctx context.Context) error {
-	opts := []func(context.Context) error{
-		rc.setGlobalVariables,
-		rc.restoreSchema,
-		rc.preCheckRequirements,
-		rc.initCheckpoint,
-		rc.restoreTables,
-		rc.fullCompact,
-		rc.cleanCheckpoints,
+	type processFunc func(context.Context) error
+	var opts []processFunc
+	if rc.cfg.CheckOnly == nil {
+		opts = []processFunc{
+			rc.setGlobalVariables,
+			rc.restoreSchema,
+			rc.preCheckRequirements,
+			rc.initCheckpoint,
+			rc.restoreTables,
+			rc.fullCompact,
+			rc.cleanCheckpoints,
+		}
+	} else {
+		opts = []processFunc{
+			rc.setGlobalVariables,
+			rc.loadSchemaForCheckOnly,
+			rc.preCheckRequirements,
```

Comment:
Should the data sample check parameters also affect the precheck data percentage. I think if I set the percentage to 100% and the both precheck and sample check passed, I will expect a following import should succeed. But, a source line with invalid data type after 1000 lines will still cause import failure, with is not expected and anti-intuition.  @sunzhaoyang  what do you think?

Reply:
BTW, do we allow set `check-requirements = false` in check-only mode?

Reply:
> BTW, do we allow set check-requirements = false in check-only mode?

yes, check-only doesn't affect logic which depends on this flag

Reply:
> Should the data sample check parameters also affect the precheck data percentage. I think if I set the percentage to 100% and the both precheck and sample check passed, I will expect a following import should succeed. But, a source line with invalid data type after the first 1000 lines will still cause import failure, with is not expected and anti-intuition. @sunzhaoyang what do you think?

No. As my opinion sample check parameters should not affect the precheck data percentage. If there are some errors occurs maybe we should category them to simple check.  

I don't want to make it complicated to understand.  So the `invalid data type` you mentioned I wish it could be reported by sample check.